user-interface/src/main/java/life/qbic/datamanager/views/projects/create/ProjectDesignLayout.java

```diff
@@ -97,16 +128,41 @@ private void initCodeGenerationButton() {
     generateCodeButton.setTooltipText("Generate Project ID");
     generateCodeButton.addThemeVariants(ButtonVariant.LUMO_ICON);
     generateCodeButton.addClickListener(
-        buttonClickEvent -> codeField.setValue(ProjectCode.random().value()));
+        //Only non-used project code should be provided
+        buttonClickEvent -> {
+          String newProjectCode = ProjectCode.random().value();
+          boolean isProjectCodeDuplicated = !isProjectCodeUnique(newProjectCode);
+          int retries = 0;
+          final int MAX_NUMBER_OF_RETRIES = 20;
+          while (isProjectCodeDuplicated) {
+            //Ensure that we have an exit condition and the while loop does not iterate endlessly if most project codes are taken
+            if (retries > MAX_NUMBER_OF_RETRIES) {
+              break;
+            }
+            newProjectCode = ProjectCode.random().value();
+            isProjectCodeDuplicated = isProjectCodeUnique(newProjectCode);
+            retries++;
+          }
+          codeField.setValue(newProjectCode);
```

Comment:
As this is a very complex lamda, please extract a method here.

Reply:
done! 🧠